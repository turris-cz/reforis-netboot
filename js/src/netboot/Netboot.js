/*
 * Copyright (C) 2020 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

import React from "react";
import PropTypes from "prop-types";

import Devices from "./Devices";

Netboot.propTypes = {
    ws: PropTypes.object.isRequired,
};

export default function Netboot({ ws }) {
    return (
        <>
            <h1>{_("Netboot")}</h1>
            <p>{_("Manage devices which can be booted from this router through network.")}</p>
            <Devices ws={ws} />
        </>
    );
}
