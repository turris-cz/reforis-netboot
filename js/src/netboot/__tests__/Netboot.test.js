/*
 * Copyright (C) 2020 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

import React from "react";
import { render } from "foris/testUtils/customTestRender";
import { WebSockets } from "foris";

import Netboot from "../Netboot";

describe("<Netboot />", () => {
    it("should render component", () => {
        const webSockets = new WebSockets();
        const { container } = render(<Netboot ws={webSockets} />);
        expect(container).toMatchSnapshot();
    });
});
