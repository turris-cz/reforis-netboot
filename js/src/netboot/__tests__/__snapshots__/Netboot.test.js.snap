// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Netboot /> should render component 1`] = `
<div>
  <h1>
    Netboot
  </h1>
  <p>
    Manage devices which can be booted from this router through network.
  </p>
  <div
    class="spinner-wrapper my-3 text-center"
  >
    <div
      class="spinner-border"
      role="status"
    >
      <span
        class="sr-only"
      />
    </div>
  </div>
</div>
`;
